DONKEY-561 Make the sandbox tree collapse button tabbable

kailin-limble opened the pull request:
to conform to accessibility standards

https://limble.atlassian.net/browse/DONKEY-561

kailin-limble committed:
DONKEY-561 Make the sandbox tree collapse button tabbable

to conform to accessibility standards

slcwebguy commented on projects/sandbox/src/app/collapsible/collapsible.component.html:
```diff
@@ -1,6 +1,6 @@
 <tree-node>
    <div class="container">
-      <div class="collapse-button" (click)="toggle()">
+      <button class="collapse-button" (click)="toggle()">
```
```suggestion
      <div role="button" class="collapse-button" (click)="toggle()">
```

question: could you, maybe, do this instead and remove the need for the CSS updates?

slcwebguy replied:
Never mind. That probably wouldn't be good cause the element would still not be focusable or respond correctly to key presses.

kailin-limble replied:
yeah I belieev `button` is semantically correct. I see some places in webapp where the other teams are using 

`<div role="button" tabIndex="0" (click)... (keyup.Enter)... (keyup.Space)... >`

when they could be using `button`

slcwebguy commented:
question: did you check these updates across browsers by chance? What you have looks like it should work. I'm just wondering if maybe there are some different browser-specific implementations of `unset` and/or `revert`. I've not used those before myself.

kailin-limble commented:
> question: did you check these updates across browsers by chance? What you have looks like it should work. I'm just wondering if maybe there are some different browser-specific implementations of `unset` (and/or `revert`) which would be problematic. I've not used those before myself.

standard on all browsers except IE11

cheefbird committed:
Merge branch 'main' into DONKEY-561---Tab-to-select-next

cheefbird committed:
Merge branch 'main' into DONKEY-561---Tab-to-select-next

cheefbird committed:
fix: improve a11y of collapse button

- add type, aria-expanded, and aria-label attributes
- mark icons as aria-hidden
- improve focus-visible styling with solid outline and offset
